docs/tutorials/vagrant-discovery.rst

```diff
@@ -0,0 +1,370 @@
+
+``[Note]`` You need to execute below command in the dirctory which contains ``Vagantfile`` on host.If you are in vagrant box, you need to execute ``exit`` command to exit box and then execute below command. 
+
+.. code::
+  
+  sudo vagrant up quanta_d51
+
+You can execute command on host to check whether quanta_d51 vnode is up successfully. If the status of quanta_d51 vnode is ``running``, quanta_d51 is up successfully.
+
+.. code::
+
+  sudo vagrant status
+
+If you are interested in quanta_d51 vnode, you can execute the below command on host to login in quanta_d51 vnode with password: **root**. Then execute the command ``exit`` to enter host.
+
+.. code::
+
+  sudo vagrant ssh quanta_d51
+
+2. On the Windows desktop, double-click the UltraVNC Viewer, and connect to <IP>:15901,to view the PXE progress to boot the microkernel. The <IP> should be set to the node ip where ``quanta_d51 vnode`` is set up by using command **"Vagrant up quanta_d51"**. For example, If you set up environmet on windows, you just need to set <IP> to ``localhost``. If you set up environment on Linux, you just need to get IP by ``ifconfig eth0`` and then replace <IP>.
+
+.. image:: ../_static/node_discovery_1.png
+     :align: center
+
+.. image:: ../_static/node_discovery_2.png
+     :align: center
+
+3. The vNode console pauses for about 1 minute, to catalog the vnode information.The vNode will reboot after cataloging finishes. This reboot indicates that the discovery workflow is completed.
+
+.. image:: ../_static/node_discovery_finish.png
+     :align: center
+
+4.Retrieve the nodes by typing the following RackHD API to discover the node.
+
+.. code::
+   
+   vagrant@rackhd:~$ curl localhost:8080/api/2.0/nodes | jq '.'
+
+You can see one or more vnodes whose type is "enclosure" or "compute".
+
+.. image:: ../_static/curl_nodes_info.png
+    :align: center
+
+NodeId
+-------
+
+NodeId is the unique Identity of a node in RackHD. List all the compute type nodes being discovered on the rackhd-server SSH console by typing the following command. Append "?type=compute" as a query string.
+
+You will focus on compute type nodes in this module
+
+.. code::
+
+  vagrant@rackhd:~$ curl 127.0.0.1:8080/api/2.0/nodes?type=compute | jq '.'
+
+In the following json output, the compute node ID is ``58b660116d20657f0c5d6466``. You will take it as a variable named <node_id> in the following module.
+
+**Note**: The node_id varies for different nodes. Even for the same node, the Node ID changes if the RackHD database is being cleaned and the node rediscovered.
+
+Do not use the example ``58b660116d20657f0c5d6466`` in your vLab. Use the displayed Node ID in your lab.
+
+
+Retrieve Catalogs
+-----------------
+
+Catalogs are described as the following:
+
+- Free form data structures with information about the nodes
+- Pluggable mechanisms for adding new catalogers for additional data
+- JSON documents stored in MongoDB
+
+Examples of catalog sources include the following:
```

Comment:
Drive smart information;
driveId catalog including system identified drive information.

The lines are added